era5cli/fetch.py

```diff
@@ -267,6 +280,34 @@ def _check_variable(self, variable):
                 "Invalid variable name: {}".format(variable)
             )
 
+    def _check_area(self):
+        """Confirm that area parameters are correct."""
+        (N, W, S, E) = self.area
+        if not (-90 <= N <= 90
+                and -90 <= S <= 90
+                and -180 <= W <= 180
+                and -180 <= E <= 180
+                and N > S
+                and W != E
+                ):
+            raise ValueError(
+                "Provide coordinates as N W S E, or ymax xmin ymin xmax. "
+                "x must be in range -180,+180 and y must be in range -90,+90."
+            )
+
+    def _round_area(self):
+        area = [round(coord, ndigits=2) for coord in self.area]
+        return area
+
+    def _parse_area(self):
+        """Parse area parameters to accepted coordinates."""
+        self._check_area()
+        area = self._round_area()
+        if self.area != area:
```

Comment:
Good point! I did think about that, although I had the rounding step as part of the _parse_area. I ended up pulling the rounding function out of there, and into its separate function, because I needed to call that step twice; once for the request and once for the filename. Now that this is removed (following your comment below), I pulled that step back into the _parse_area function.
It loses some readability, but it's minimal.